### DISCUSSION TITLE
Bricked HH40V

### DISCUSSION THREAD

**Discussion by Draggon** [froonix/HH40V#6] [General]
Hi

Looks like I've managed to partialy bricked my HH40V.
Story is - successfully installed OpenWrt RC2, configured - all was fine.
Than few days later after restart it and never came up (no WiFi / LAN up) only LTE module was working.
I went through TFTP recovery procedure - all was fine.
When logged into WEB there was notification about firmware updated - I confirmed and after restart looks like something went wrong and LTE module was down (no LED) and UI.

I still can SSH to router and LTE module.
I'v tried /etc/init.d/fota_progress start few times - it looks ok but at the and LTE module is down (off?).
My router is branded TMobile Magneta and this may be source of the issue - some version mismatch?

Below some parts of cat /proc/kmsg

<5>[    7.104089] UBIFS (ubi4:0): background thread "ubifs_bgt4_0" started, PID 228
<5>[    7.138858] UBIFS (ubi4:0): recovery needed
<5>[    7.220283] UBIFS (ubi4:0): recovery completed
<5>[    7.223777] UBIFS (ubi4:0): UBIFS: mounted UBI device 4, volume 0, name "jrd_vol"
<5>[    7.231204] UBIFS (ubi4:0): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
<5>[    7.241088] UBIFS (ubi4:0): FS size: 71995392 bytes (68 MiB, 567 LEBs), journal size 3555328 bytes (3 MiB, 28 LEBs)
<5>[    7.251502] UBIFS (ubi4:0): reserved for root: 3400519 bytes (3320 KiB)
<5>[    7.258103] UBIFS (ubi4:0): media format: w4/r0 (latest is w4/r0), UUID B0923CD2-B8C1-46AA-B98F-9CDA99207355, small LPT model
<12>[    7.307723] /dev/loop0 head is mkdosfs
<6>[    7.375965] lun0: jrd test lun blksize=512
<6>[    7.378953] lun0: jrd test lun blkbits=9<6>[    7.518574] subsys-restart: __subsystem_get(): Changing subsys fw_name to modem
<6>[    7.534439] pil-q6v5-mss 4080000.qcom,mss: modem: loading from 0x82a00000 to 0x86200000
<4>[    7.579640] pil-q6v5-mss 4080000.qcom,mss: Debug policy not present - msadp. Continue.
<6>[    7.594530] pil-q6v5-mss 4080000.qcom,mss: Loading MBA and DP (if present) from 0x87900000 to 0x87a00000
<6>[    7.650294] pil-q6v5-mss 4080000.qcom,mss: MBA boot done
<3>[    7.658441] pil-q6v5-mss 4080000.qcom,mss: MBA returned error -3 for headers
<3>[    7.670854] RMB_MBA_IMAGE: 87900000
<3>[    7.673312] RMB_PBL_STATUS: 00000001
<3>[    7.680958] RMB_MBA_COMMAND: 00000000
<3>[    7.683589] RMB_MBA_STATUS: fffffffd
<3>[    7.691653] RMB_PMI_META_DATA: 87850000
<3>[    7.697661] RMB_PMI_CODE_START: 00000000
<3>[    7.700552] RMB_PMI_CODE_LENGTH: 00000000
<3>[    7.709061] RMB_PROTOCOL_VERSION: 00000000
<3>[    7.712124] RMB_MBA_DEBUG_INFORMATION: 00000007
<4>[    7.824332] pil-q6v5-mss 4080000.qcom,mss: Port c8b08000 halt timeout
<3>[    7.829919] pil-q6v5-mss 4080000.qcom,mss: modem: **Invalid firmware metadata**
<3>[    7.844534] M-Notify: General: 7
<3>[    7.846834] M-Notify: General: 5
<3>[    7.853009] kathy:msm_ipc_router_ioctl:cmd:-1072381182,port_ptr->type:0
<12>[    8.181172] QTI:start
<5>[    8.523820] random: sshd urandom read with 53 bits of entropy available
<3>[    8.666414] kathy:msm_ipc_router_ioctl:cmd:-2146385147,port_ptr->type:0
<3>[    8.676134] kathy:msm_ipc_config_sec_rules:ret:-22,port_ptr->type:0
<3>[    8.682595] kathy:msm_ipc_router_close_port:port_ptr->type:3
<12>[   12.199004] QCMAP:start
<3>[   12.358343] kathy:msm_ipc_router_ioctl:cmd:-2147171580,port_ptr->type:0
<12>[   12.390156] USB QCMAP NL IOCTL Snd GETNEIGH Succ
<4>[   12.422604] ------------[ cut here ]------------
<4>[   12.426484] WARNING: CPU: 0 PID: 437 at /home/yunhua.zhao/HH40/ER/MDM9207/apps_proc/oe-core/build/tmp-glibc/work-shared/mdm9607/kernel-source/net/netfilter/nf_queue.c:35 nf_register_queue_handler+0x28/0x38()
<4>[   12.444669] Modules linked in:
<4>[   12.447640] CPU: 0 PID: 437 Comm: sh Tainted: G        W      3.18.20 #1
<4>[   12.454525] [<c00132ac>] (unwind_backtrace) from [<c0011460>] (show_stack+0x10/0x14)
<4>[   12.462056] [<c0011460>] (show_stack) from [<c001c6ac>] (warn_slowpath_common+0x68/0x88)
<4>[   12.470198] [<c001c6ac>] (warn_slowpath_common) from [<c001c75c>] (warn_slowpath_null+0x18/0x20)
<4>[   12.478949] [<c001c75c>] (warn_slowpath_null) from [<c04b064c>] (nf_register_queue_handler+0x28/0x38)
<4>[   12.488155] [<c04b064c>] (nf_register_queue_handler) from [<c04c95c4>] (proc_sip_segment+0x38/0x54)
<4>[   12.497155] [<c04c95c4>] (proc_sip_segment) from [<c0129710>] (proc_sys_call_handler+0x98/0xb4)
<4>[   12.506355] [<c0129710>] (proc_sys_call_handler) from [<c012973c>] (proc_sys_write+0x10/0x18)
<4>[   12.514581] [<c012973c>] (proc_sys_write) from [<c00d95d0>] (vfs_write+0xb8/0x170)
<4>[   12.521860] [<c00d95d0>] (vfs_write) from [<c00d9a40>] (SyS_write+0x40/0x80)
<4>[   12.528940] [<c00d9a40>] (SyS_write) from [<c000dd00>] (ret_fast_syscall+0x0/0x38)
<4>[   12.536473] ---[ end trace c1e3312482b205e9 ]---`

apr_register: Modem is not Up

<12>[  114.699759] restart core_app
<6>[  115.185665] store_jrd_panic_reboot: jrd_panic_reboot(before, after)=(1,1)
<12>[  115.239484] core_app:init start
<3>[  115.243104] kathy:msm_ipc_router_ioctl:cmd:-2147171580,port_ptr->type:0

Any ideas what to do ? ;)

_________
Update:

Looks like there is an issue with updating firmware:
FOTA>>>>>>>>handle_deltaupdate_status entered!
FOTA>>>>>>>>number of files in zip is 1078
FOTA>>>>>>>> update_status = 0
FOTA>>>>>>>> modem_update_status = 0
safe-rm: wei rm /cache/temp/mba.b00_flag /cache/temp/mba.b00_flag
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/mba.b01_flag /cache/temp/mba.b01_flag
safe-rm: wei rm /cache/temp/mba.b01.diff /cache/temp/mba.b01.diff
safe-rm: wei rm /cache/temp/mba.b02_flag /cache/temp/mba.b02_flag
safe-rm: wei rm /cache/temp/mba.b03_flag /cache/temp/mba.b03_flag
safe-rm: wei rm /cache/temp/mba.b04_flag /cache/temp/mba.b04_flag
safe-rm: wei rm /cache/temp/mba.b05_flag /cache/temp/mba.b05_flag
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/mba.mbn_flag /cache/temp/mba.mbn_flag
safe-rm: wei rm /cache/temp/mba.mbn.diff /cache/temp/mba.mbn.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/mba.mdt_flag /cache/temp/mba.mdt_flag
safe-rm: wei rm /cache/temp/mba.mdt.diff /cache/temp/mba.mdt.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b00_flag /cache/temp/modem.b00_flag
safe-rm: wei rm /cache/temp/modem.b00.diff /cache/temp/modem.b00.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b01_flag /cache/temp/modem.b01_flag
safe-rm: wei rm /cache/temp/modem.b01.diff /cache/temp/modem.b01.diff
safe-rm: wei rm /cache/temp/modem.b02_flag /cache/temp/modem.b02_flag
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b03_flag /cache/temp/modem.b03_flag
safe-rm: wei rm /cache/temp/modem.b03.diff /cache/temp/modem.b03.diff
safe-rm: wei rm /cache/temp/modem.b04_flag /cache/temp/modem.b04_flag
safe-rm: wei rm /cache/temp/modem.b05_flag /cache/temp/modem.b05_flag
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b06_flag /cache/temp/modem.b06_flag
safe-rm: wei rm /cache/temp/modem.b06.diff /cache/temp/modem.b06.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b07_flag /cache/temp/modem.b07_flag
safe-rm: wei rm /cache/temp/modem.b07.diff /cache/temp/modem.b07.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b08_flag /cache/temp/modem.b08_flag
safe-rm: wei rm /cache/temp/modem.b08.diff /cache/temp/modem.b08.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b09_flag /cache/temp/modem.b09_flag
safe-rm: wei rm /cache/temp/modem.b09.diff /cache/temp/modem.b09.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b10_flag /cache/temp/modem.b10_flag
safe-rm: wei rm /cache/temp/modem.b10.diff /cache/temp/modem.b10.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b11_flag /cache/temp/modem.b11_flag
safe-rm: wei rm /cache/temp/modem.b11.diff /cache/temp/modem.b11.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b12_flag /cache/temp/modem.b12_flag
safe-rm: wei rm /cache/temp/modem.b12.diff /cache/temp/modem.b12.diff
safe-rm: wei rm /cache/temp/modem.b13_flag /cache/temp/modem.b13_flag
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b14_flag /cache/temp/modem.b14_flag
safe-rm: wei rm /cache/temp/modem.b14.diff /cache/temp/modem.b14.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b15_flag /cache/temp/modem.b15_flag
safe-rm: wei rm /cache/temp/modem.b15.diff /cache/temp/modem.b15.diff
safe-rm: wei rm /cache/temp/modem.b16_flag /cache/temp/modem.b16_flag
safe-rm: wei rm /cache/temp/modem.b17_flag /cache/temp/modem.b17_flag
safe-rm: wei rm /cache/temp/modem.b18_flag /cache/temp/modem.b18_flag
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b19_flag /cache/temp/modem.b19_flag
safe-rm: wei rm /cache/temp/modem.b19.diff /cache/temp/modem.b19.diff
**cp: can't create '/firmware/image/modem.b20': File exists
chmod: /firmware/image/modem.b20: Read-only file system**
safe-rm: wei rm /cache/temp/modem.b20_flag /cache/temp/modem.b20_flag
safe-rm: wei rm /cache/temp/modem.b20 /cache/temp/modem.b20
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b21_flag /cache/temp/modem.b21_flag
safe-rm: wei rm /cache/temp/modem.b21.diff /cache/temp/modem.b21.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b22_flag /cache/temp/modem.b22_flag
safe-rm: wei rm /cache/temp/modem.b22.diff /cache/temp/modem.b22.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b23_flag /cache/temp/modem.b23_flag
safe-rm: wei rm /cache/temp/modem.b23.diff /cache/temp/modem.b23.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.b24_flag /cache/temp/modem.b24_flag
safe-rm: wei rm /cache/temp/modem.b24.diff /cache/temp/modem.b24.diff
safe-rm: wei rm -rf -rf
safe-rm: wei rm /cache/temp/modem.mdt_flag /cache/temp/modem.mdt_flag
FOTA>>>>>>>>modem_update end!

**Comment by boolinek** [comment]
I had a similar problem - slightly different scenario, but the effects were the same. I managed to get my HH40V working though. To do this, you need to:
1. reinstall OpenWRT (it does not have a firewall blocking access to the modem)
2. you also need the stock firmware package for [HH40V](https://alcatelfirmware.com/alcatel-onetouch-hh40v)
3. unpack the package and go to /Firmare/FotaImage
4. from the contents of this directory - except for the files efs.mbn, otacerts.zip and VER_020024 - create an update.zip archive
5. connect via www to HH40V and change the WAN interface device to usb0 (if it is not there, restart the box), check if it has an IP address, if not, configure it manually (IP: 192.168.225.111, MASK 255.255.255.0, GATEWAY/DNS: 192.168.225.1)
6. connect a Linux computer to HH40V
7. connect to the modem via SSH and remove write protection of /extern and /firmware directories (`mount -o remount,rw /dev/loop0` `mount -o remount,rw /dev/ubi1_0`)
8. run Midnight Commander on it and connect to the modem via shell (F9 --> Right --> Connect via shell --> `sh://root@192.168.225.1`
8. copy the created update.zip file to the /extern/fota directory in modem
9. run the fota_progress script (`/etc/init.d/fota_progress start`)
10. after the automatic restart of the box the modem should light up
11. all that remains is to install the SIM card and configure OpenWRT

**Reply by tosiara** [reply]
That worked for me!!! Thanks!

**Comment by tosiara** [comment]
Got the same brick.
I successfully installed OpenWRT, but for some reason experienced low speed. Decided to revert to stock to do the baseline speed test. TFTP recovery went well, but I saw an update in the web UI. God damn it, I installed that update and ended up in the same brick as the OP :cry:

**Comment by AndreasUfert** [comment]
Same problem here. Installed openwrt, everything fine except I didn't know how to receive SMS messages from my provider (used as 2FA on their website). So I went back to stock, saw the update and - bricked.

ssh in Firmware still accessible, yet no web interface (webs is stuck) and modem seems to be dead. Following the instructions above it fails when ssh'ing into the modem:

ssh: Connection to root@192.168.225.1:22 exited: Connect failed: Host is unreachable

Any clue how to reflash the modem?

**Reply by boolinek** [reply]
In the original Firmware, access to the modem is blocked by the firewall, so it is necessary to reinstall OpenWRT (point 1 of the instructions). You must follow the instructions step by step.

**Reply by AndreasUfert** [reply]
Sorry, should have given it more precisely:

- no usb0 device in stock firmware (="dead")
- in openwrt 24.10.2, there is a device usb0, however even with an IP address manually set, I can't ssh into the modem

**Reply by boolinek** [reply]
Unfortunately I can't help you - I no longer have the HH40V.

**Reply by tosiara** [reply]
`Host is unreachable` comes from your system, you haven't set local IP properly, check with `ip a` command, what does show?
